unit_tests/transfer_channel/mock_stream.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef __MOCK_STREAM_HPP
+#define __MOCK_STREAM_HPP
+
+#include "cubstream.hpp"
+
+class mock_stream : public cubstream::stream
```

Comment:
Since mock_stream may be useful in multiple tests, I think it is best to move it to shared unit test library.